src/grouping_map.rs

```diff
@@ -461,7 +461,7 @@ impl<I, K, V> GroupingMap<I>
                 Some(MinMaxResult::MinMax(min, max)) => {
                     if compare(key, &val, &min) == Ordering::Less {
                         MinMaxResult::MinMax(val, max)
-                    } else if compare(key, &val, &max) == Ordering::Greater {
+                    } else if compare(key, &val, &max) != Ordering::Less {
```

Comment:
This seems like an odd change. Doesn't this render the `else` branch completely unreachable?

Reply:
The two `if`s are different. The first compares `val` with `min`, the second with `max`.
The final `else` should be executed when `val` is in between `min` and `max`.

The commit should change the behaviour when `val` is equal to `max`.
Before it was executed the `else` branch meaning `max` was keeped, now the second `if` branch is executed and `val` becomes the new maximum.

This was done to be consistent with `std::iter::max` that states that "If several elements are equally maximum, the last element is returned.". Since `val` comes after `max` it should be considered the maximum.

I think I should add some tests and maybe write this behaviour in the docs to make it clearer.